pkg/controller/extendeddaemonsetreplicaset/ddaemonsetreplicaset_controller.go

```diff
@@ -419,15 +425,20 @@ func (r *ReconcileExtendedDaemonSetReplicaSet) getOldDaemonsetPodList(ds *datado
 		// Error reading the object - requeue the request.
 		return nil, err
 	}
-	var podSelector labels.Set
+	podListOptions := []client.ListOption{}
 	if oldDaemonset.Spec.Selector != nil {
-		podSelector = labels.Set(oldDaemonset.Spec.Selector.MatchLabels)
-	}
-	podListOptions := []client.ListOption{
-		client.MatchingLabelsSelector{
-			Selector: podSelector.AsSelectorPreValidated(),
-		},
+		selector, err2 := utils.LabelSelector2LabelSelector(log, oldDaemonset.Spec.Selector)
```

Comment:
is the error var name intentional?

Reply:
It was to fix a linter error complaining about variable shadowing.
But it’s clearer if I remove the named output parameter.
So, I changed to use `err` now.